### ISSUE TITLE
`<script type="module">` should support top-level-await

### ISSUE THREAD

**Issue by loynoir** [BenoitZugmeyer/eslint-plugin-html#247]
## Setup configuration

- ESLint version: v8.48.0
- eslint-plugin-html version: 7.1.0
- NodeJS version: v20.5.1
- Operating System name and version:
- ESLint configuration (.eslintrc):

```json
      {
        "files": [
          "*.html"
        ],
        "globals": {
          "console": "readonly",
          "Promise": "readonly"
        },
        "plugins": [
          "html"
        ],
        "parserOptions": {
          "ecmaVersion": "latest"
        },
        "extends": [
          "eslint:recommended"
        ]
      }
```

Optional: custom plugin for debugging

```js
      Program(node) {
        console.log(node.sourceType)
```

- Editor name and version:
- Plugin name and version:

## Aditional context

```html
    <script>
      const a = 42
    </script>
    <script type="module">
      const b = await Promise.resolve(42)
    </script>
```

**Actual**

eslint log

```txt
  ...  error  'a' is assigned a value but never used   no-unused-vars
  ...  error  Parsing error: Unexpected token Promise
```

plugin log

- With TLA

```txt
script
```

the second script tag not reach plugin

- Without TLA

```txt
script
script
```

the second script tag reach plugin with wrong `node.sourceType`

**Expected**

eslint log

```
  ...  error  'a' is assigned a value but never used  no-unused-vars
  ...  error  'b' is assigned a value but never used  no-unused-vars
```

Plugin log

```txt
script
module
```

**Comment by BenoitZugmeyer** [comment]
For simplicity's sake, this plugin relies on the ESLint `parserOption.sourceType` configuration to consider inline scripts as "traditional" or "module" (see [doc](https://github.com/BenoitZugmeyer/eslint-plugin-html?tab=readme-ov-file#multiple-scripts-tags-in-a-html-file)). It does not consider `type="module"` as "module" by default, and you can't have a mix of traditional and "module" inline scripts in a single HTML file.

This might change in the future, see https://github.com/BenoitZugmeyer/eslint-plugin-html/issues/139
